### PR TITLE
[Testcase Refactoring] Decouple inductor compile collectives tests from hardware assumptions

### PR DESCRIPTION
**Pull request by ll-pen**
I reviewed the implementation and test results and confirm that the quoted description matches this change. Codex helped draft the quoted text.

> ## Summary
>
> Decouple `test_inductor_compile_collectives.py` from implicit hardware assumptions by adding explicit hardware classification labels.
>
> Part of #185590.
>
> ## Changes
>
> - Add corresponding `HardwareClassification` labels to test classes.
> - Preserve existing parameterization, backend admission, expected failures, and test behavior.
>
> ## Test Plan
>
> Ran:
>
> ```bash
> python -m py_compile test/distributed/test_inductor_compile_collectives.py
> lintrunner -a test/distributed/test_inductor_compile_collectives.py
> git diff --check
> ```
>
> CI:
>
> - Run the modified test file through its existing PyTorch test jobs.
> - Run the Inductor workflows selected by `ciflow/inductor`.

cc @wjlFlyer @pjyFight @FuDdd @fffrog @lyriexs666 @jishuangfeng @JamesD18 @JiasenTian

### CODE REVIEW

File: test/distributed/test_inductor_compile_collectives.py [hunk 1]
```diff
@@ -34,6 +38,8 @@ def _make_fx_with_allreduce():
 
 
 class TestInductorCompileCollectives(TestCase):
+    hw_classification = HardwareClassification.CPU
+
```
Comment (lizhihai137) [inline test/distributed/test_inductor_compile_collectives.py:45]: The classification of class TestInductorCompileCollectives should be GENERIC?

**Review by lizhihai137 — CHANGES_REQUESTED** [review]

File: test/distributed/test_inductor_compile_collectives.py (hunk 1 above)
Comment (ll-pen) [reply test/distributed/test_inductor_compile_collectives.py:45]: done

**Review by lizhihai137 — APPROVED** [review]

File: test/distributed/test_inductor_compile_collectives.py [hunk 2]
```diff
@@ -33,7 +39,10 @@ def _make_fx_with_allreduce():
     return make_fx(_f)(torch.ones(4))
 
 
+@requires_gloo()
```
Comment (KarhouTam) [inline test/distributed/test_inductor_compile_collectives.py:42]: Why added this?

Comment (ll-pen) [reply test/distributed/test_inductor_compile_collectives.py:42]: I added `requires_gloo()` because `setUp()` explicitly initializes the process group with `backend="gloo"`, while `dist.is_available()` only confirms that distributed support is built and does not guarantee that the Gloo backend is available. The decorator makes this existing runtime dependency explicit and skips the tests cleanly on builds without Gloo.

This is orthogonal to the `GENERIC` classification: `GENERIC` describes the hardware semantics being tested, while Gloo remains an optional process-group dependency used by the fixture.

My understanding is that this makes the pre-existing dependency explicit. That said, it is not required for the classification change itself and was not present in the original code. If you prefer keeping this PR strictly classification-only, I am also fine removing it and handling the Gloo availability check separately.
